autode/calculation.py

```diff
@@ -1,312 +1,451 @@
+                          f'{get_available_solvent_names(method)}')
 
-        Arguments:
-            molecule (mol obj): the molecule being calculated
-        """
-        active_atoms = set()
-        for bond in self.bond_ids_to_add:
-            active_atoms.add(bond[0])
-            active_atoms.add(bond[1])
+    if molecule.solvent is None:
+        logger.info('Calculation is in the gas phase')
+        return None
 
-        core_atoms = set()
-        for active_atom in active_atoms:
-            bonded_atoms = molecule.graph.neighbors(active_atom)
-            core_atoms.add(active_atom)
+    solvent_name = getattr(molecule.solvent, method.name)
+    if solvent_name is None:
+        raise SolventUnavailable(message=available_solvents)
 
-            for bonded_atom in bonded_atoms:
-                core_atoms.add(bonded_atom)
+    return solvent_name
 
-        self.core_atoms = list(core_atoms)
-        return None
 
-    def _check_calc(self):
+class Calculation:
+
+    def _check_molecule(self):
+        """Ensure the molecule has the required attributes"""
         assert hasattr(self.molecule, 'n_atoms')
         assert hasattr(self.molecule, 'atoms')
         assert hasattr(self.molecule, 'mult')
         assert hasattr(self.molecule, 'charge')
         assert hasattr(self.molecule, 'solvent')
 
-        # Ensure the distance constraint dictionary is of the correct format
-        if self.distance_constraints is not None:
-            assert type(self.distance_constraints) is dict
-            assert all(len(key) == 2 for key in self.distance_constraints.keys())
+        # The molecule must have > 0 atoms
+        if self.molecule.atoms is None or self.molecule.n_atoms == 0:
+            logger.error('Have no atoms. Can\'t form a calculation')
+            raise NoInputError
 
-        # Ensure the point charges are given as a list of PointCharge objects
-        if self.point_charges is not None:
-            assert type(self.point_charges) is list
-            assert all(type(pc) is PointCharge for pc in self.point_charges)
+    def _get_energy(self, e=False, h=False, g=False, force=False):
+        """
+        Get the energy from a completed calculation
 
-        # Key attributes that need to be lists or None
-        assert self.cartesian_constraints is None or type(self.cartesian_constraints) is list
-        assert self.keywords_list is None or type(self.keywords_list) is list
+        Keyword Arguments:
+            e (bool): Return the potential energy (E)
+            h (bool): Return the enthalpy (H) at 298 K
+            g (bool): Return the Gibbs free energy (G) at 298 K
+            force (bool): Return the energy even if the calculation errored
 
-    def get_energy(self):
-        logger.info(f'Getting energy from {self.output_filename}')
-        if self.terminated_normally:
-            return self.method.get_energy(self)
+        Returns:
```

Comment:
the three get energies seems like it repeats itself a bit, but I;'m sure it will become apparent later on why this is

Reply:
I didn't know the best way to do this - it was this or repeat the logging. Will leave it for the time being